### PR TITLE
Fix BindableFloat.IsDefault returning true for non-default values

### PR DESCRIPTION
**Pull request by iiSaLMaN**
- Takes 50% of the precision to ensure no-underflow for values near to default (for example, `Value = 1.01f`, `Default = 1.00f`, `Precision = 0.01f`)

Fixes *"no revert button for UI-scale 1.01x"* part of ppy/osu#5940

### CODE REVIEW

**Comment by peppy** [comment]
Missing tests.

**Comment by iiSaLMaN** [comment]
> Missing tests.

Resolved.

File: osu.Framework/Bindables/BindableFloat.cs [hunk 1]
```diff
@@ -8,7 +8,8 @@ namespace osu.Framework.Bindables
 {
     public class BindableFloat : BindableNumber<float>
     {
-        public override bool IsDefault => Math.Abs(Value - Default) < Precision;
+        // Take 50% of the precision to ensure the value doesn't underflow and return true for non-default values.
+        public override bool IsDefault => Math.Abs(Value - Default) < (Precision / 2);
```
Comment (smoogipoo) [inline osu.Framework/Bindables/BindableFloat.cs]: Probably should do the same for BindableDouble.

**Comment by peppy** [comment]
Regressed (see https://github.com/peppy/osu-framework/tree/broken-default)

**Comment by bdach** [comment]
This is most likely due to `float.Epsilon` underflowing to zero upon division by two, since that's the smallest fractional value that can be represented in that data type. I'd wager a `Math.Max(precision, float.Epsilon)` would fix this. Same thing applies for `double`, of course.

**Comment by iiSaLMaN** [comment]
> The regression mentioned above is most likely due to `float.Epsilon` underflowing to zero upon division by two, since that's the smallest fractional value that can be represented in that data type. I'd wager a `Math.Max(precision, float.Epsilon)`, or setting the default to `2 * float.Epsilon` would fix this. Same thing applies for `double`, of course.

i think using `Precision`'s epsilon would be better, no?

**Comment by bdach** [comment]
True, that's probably a saner default in most cases.

**Pull request merged**